File: app/src/main/java/com/example/calculatorBMI/Quiz.java
package com.example.calculatorBMI;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class Quiz extends AppCompatActivity implements View.OnClickListener {

    Button button1, button2, button3, button4;
    private Question question= new Question();
    TextView question_text;
    private String answer;
    private int questionlength=question.questions.length;
    Random random;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_quiz);
        random = new Random();
        button1=(Button)findViewById(R.id.button1);
        button1.setOnClickListener(this);

        button2=(Button)findViewById(R.id.button2);
        button2.setOnClickListener(this);

        button3=(Button)findViewById(R.id.button3);
        button3.setOnClickListener(this);

        button4=(Button)findViewById(R.id.button4);
        button4.setOnClickListener(this);
        question_text=(TextView)findViewById(R.id.questions);
        NextQuestion(random.nextInt(questionlength));

    }

    @Override
    public void onClick(View v) {
        switch (v.getId())
        {
            case R.id.button1:
                if (button1.getText()==answer){
                    Toast.makeText(this,"Prawidłowo!", Toast.LENGTH_LONG).show();
                    NextQuestion(random.nextInt(questionlength));
                }
                break;
            case R.id.button2:
                if (button2.getText()==answer){
                    Toast.makeText(this,"Prawidłowo!", Toast.LENGTH_LONG).show();
                    NextQuestion(random.nextInt(questionlength));
                }
                break;
            case R.id.button3:
                if (button3.getText()==answer){
                    Toast.makeText(this,"Prawidłowo!", Toast.LENGTH_LONG).show();
                    NextQuestion(random.nextInt(questionlength));
                }
                break;
            case R.id.button4:
                if (button4.getText()==answer){
                    Toast.makeText(this,"Prawidłowo!", Toast.LENGTH_LONG).show();
                    NextQuestion(random.nextInt(questionlength));
                }
                break;
        }

    }

    private void NextQuestion(int x) {
        question_text.setText(question.getQuestion(x));
        button1.setText(question.getchoice1(x));
        button2.setText(question.getchoice2(x));
        button3.setText(question.getchoice3(x));
        button4.setText(question.getchoice4(x));
        answer=question.getCorrectAnswer(x);

    }
}

File: app/src/main/java/com/example/calculatorBMI/Quiz.kt
package com.example.calculatorBMI

import androidx.appcompat.app.AppCompatActivity
import com.example.calculatorBMI.Question
import android.widget.TextView
import android.os.Bundle
import com.example.calculatorBMI.R
import android.widget.Toast
import android.widget.EditText
import android.content.Intent
import android.view.View
import com.example.calculatorBMI.Activity2
import com.example.calculatorBMI.Quiz
import com.example.calculatorBMI.GraphActivity
import android.webkit.WebView
import android.widget.Button
import java.util.*

class Quiz : AppCompatActivity(), View.OnClickListener {
    var button1: Button? = null
    var button2: Button? = null
    var button3: Button? = null
    var button4: Button? = null
    private val question = Question()
    var question_text: TextView? = null
    private var answer: String? = null
    private val questionlength = question.questions.size
    var random: Random? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_quiz)
        random = Random()
        button1 = findViewById<View>(R.id.button1) as Button
        button1!!.setOnClickListener(this)
        button2 = findViewById<View>(R.id.button2) as Button
        button2!!.setOnClickListener(this)
        button3 = findViewById<View>(R.id.button3) as Button
        button3!!.setOnClickListener(this)
        button4 = findViewById<View>(R.id.button4) as Button
        button4!!.setOnClickListener(this)
        question_text = findViewById<View>(R.id.questions) as TextView
        NextQuestion(random!!.nextInt(questionlength))
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.button1 -> if (button1!!.text === answer) {
                Toast.makeText(this, "Prawidłowo!", Toast.LENGTH_LONG).show()
                NextQuestion(random!!.nextInt(questionlength))
            }
            R.id.button2 -> if (button2!!.text === answer) {
                Toast.makeText(this, "Prawidłowo!", Toast.LENGTH_LONG).show()
                NextQuestion(random!!.nextInt(questionlength))
            }
            R.id.button3 -> if (button3!!.text === answer) {
                Toast.makeText(this, "Prawidłowo!", Toast.LENGTH_LONG).show()
                NextQuestion(random!!.nextInt(questionlength))
            }
            R.id.button4 -> if (button4!!.text === answer) {
                Toast.makeText(this, "Prawidłowo!", Toast.LENGTH_LONG).show()
                NextQuestion(random!!.nextInt(questionlength))
            }
        }
    }

    private fun NextQuestion(x: Int) {
        question_text!!.text = question.getQuestion(x)
        button1!!.text = question.getchoice1(x)
        button2!!.text = question.getchoice2(x)
        button3!!.text = question.getchoice3(x)
        button4!!.text = question.getchoice4(x)
        answer = question.getCorrectAnswer(x)
    }
}